Assignment1/assignment2/lib/utilities.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:assignment2/users.dart';
+import 'dart:collection';
+
+Map<int, User> currentSession = Map();
+
+void nameHandler(String name) {
+  if (name.length == 0 || name.contains('[0-9]')) {
+    throw Exception(
+        "Name entered isn't valid, please enter appropriate details");
+  }
+}
+
+void ensurePositive(int number) {
+  if (number < 0) {
+    throw Exception(
+        "Number entered is not valid, please enter a positive number");
+  }
+}
+
+void ensureRollNumber(int rollNumber) {
+  if (currentSession.containsKey(rollNumber)) {
+    throw Exception(
+        "Roll Number already exists, please enter a new roll number");
+  }
+}
+
+List<String> acceptCourses(String? courses) {
+  List<String> registered = courses!.split(" ");
+  if (registered.length != 4) {
+    throw Exception("Registered courses should mandatorily be 4");
+  }
+  List<String> registered_courses = [];
+  registered.forEach((element) {
+    registered_courses.add(element.toUpperCase());
+  });
+  List<String> allowedCourses = ["A", "B", "C", "D", "E", "F"];
+  for (String element in registered_courses) {
+    if (allowedCourses.contains(element)) {
+      allowedCourses.remove(element);
+    } else {
+      throw Exception(
+          "Course Entered are wrong, please enter from the given courses");
+    }
+  }
+  return registered_courses;
+}
+
+void checkRollNumber(int rollNumber) {
+  if (!currentSession.containsKey(rollNumber)) {
+    throw Exception('Roll Number : $rollNumber does not exist in the database');
+  }
+}
+
+void loadCurrentSession(Map<int, User> map) {
+  currentSession = map;
+}
+
+SplayTreeMap<int, User> displayMap(String order, int option) {
+  int orderInt = 0;
+  if (order == 'a') {
+    orderInt = 1;
+  } else if (order == 'd') {
+    orderInt = -1;
+  } else {
+    throw Exception("Entered order for sorting is invalid");
+  }
+  SplayTreeMap<int, User> result = SplayTreeMap();
+  switch (option) {
+    case 1:
```

Comment:
These case values like 1, 2, 3, etc should be extracted as constants and those should be used here to increase the readability of the code.